securedrop_client/api_jobs/base.py

```diff
@@ -57,30 +57,22 @@ class ApiJob(QueueJob):
     '''
     failure_signal = pyqtSignal(Exception)
 
-    def __init__(self, remaining_attempts: int = DEFAULT_NUM_ATTEMPTS) -> None:
+    def __init__(self) -> None:
         super().__init__()
-        self.remaining_attempts = remaining_attempts
 
     def _do_call_api(self, api_client: API, session: Session) -> None:
         if not api_client:
             raise ApiInaccessibleError()
 
-        while self.remaining_attempts >= 1:
-            try:
-                self.remaining_attempts -= 1
-                result = self.call_api(api_client, session)
-            except (AuthError, ApiInaccessibleError) as e:
-                raise ApiInaccessibleError() from e
-            except (RequestTimeoutError, ServerConnectionError) as e:
-                if self.remaining_attempts == 0:
-                    self.failure_signal.emit(e)
-                    raise
-            except Exception as e:
-                self.failure_signal.emit(e)
-                raise
-            else:
-                self.success_signal.emit(result)
-                break
+        try:
+            result = self.call_api(api_client, session)
+        except (AuthError, ApiInaccessibleError) as e:
+            raise ApiInaccessibleError() from e
+        except Exception as e:
+            self.failure_signal.emit(e)
+            raise
+        else:
+            self.success_signal.emit(result)
```

Comment:
the effect of removing this automatic retry logic _could_ be that a user will see a notification much more often... I think we should understand (roughly) how often the request succeeds on the first try - if it often needs to be retried and succeeds on a subsequent try then we should keep this logic

Reply:
my thinking is that retrying 5 times (5 being the default) when we get a `ServerConnectionError` wouldn't result in more success. it doesn't hurt to retry 5 times back to back, but i don't think it's necessary. we could just fall back to the queue pause/resume-when-sync-can-connect-to-the-server logic. retrying 5 times when we get a `RequestTimeoutError` _could_ result in more success, but i think more commonly if the request times out after X seconds, since we don't increase that time in the next attempt, it'll probably time out again and again, so it seems to make sense to pause the queue and rely on queue retry logic.

Reply:
> but i think more commonly if the request times out after X seconds, since we don't increase that time in the next attempt, it'll probably time out again and again, so it seems to make sense to pause the queue and rely on queue retry logic.

@redshiftzero pointed out to me that we would end up notifying the user that we "Trying to reconnect to the SecureDrop server" after one attempt that times out, so even if it's not too common for this to occur, it would be worth researching first before removing the job retry logic and err on the side of caution so that we don't over-notify the user about network blips.